### PR TITLE
XWIKI-19788: Live Data panels have invalid attribute "panel"

### PR DESCRIPTION
**Pull request by michitux**
* Declare the panel property in all panels.
* Use the data from the panel instead of duplicating the information.

Jira issue: https://jira.xwiki.org/browse/XWIKI-19788

### CODE REVIEW

**Review by manuelleduc — APPROVED** [review]
Looks good, thanks.

**Comment by michitux** [comment]
I've added some basic tests. They don't test the invalid attribute but at least should ensure that everything still works. Some of them also test basic functionality of the tested panel.

File: xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/tests/unit/panels/LivedataAdvancedPanelFilter.spec.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+describe('LivedataAdvancedPanelFilter.vue', () => {
+  it('Displays the title and the icon', async () => {
+    const wrapper = initWrapper();
+    expect(wrapper.element.querySelector('.panel-heading .title')).toHaveTextContent('Filter');
+    expect(wrapper.element.querySelector('i')).toHaveTextContent('filter');
+  });
+
+  it('Displays a message when no filterable properties exist', async () => {
+    let wrapper = initWrapper();
```
Comment (manuelleduc) [inline xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/tests/unit/panels/LivedataAdvancedPanelFilter.spec.js:81]: can be a `const`

File: xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/tests/unit/panels/LivedataAdvancedPanelFilter.spec.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    expect(wrapper.element.querySelector('.text-muted')).toHaveTextContent('livedata.panel.filter.noneFilterable');
+    expect(wrapper.element.querySelector('.text-muted')).toHaveStyle({display: 'none'});
+    wrapper = initWrapper({
```
[the remark this answers is not in the document]
Comment (manuelleduc) [reply xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/tests/unit/panels/LivedataAdvancedPanelFilter.spec.js:84]: You can wrap the `describe` (see https://jestjs.io/docs/api#describename-fn) if it fits better your tests structure.

[the remark this answers is not in the document]
Comment (michitux) [reply xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/tests/unit/panels/LivedataAdvancedPanelFilter.spec.js]: I've split the test method.

File: xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/tests/unit/panels/LivedataAdvancedPanelFilter.spec.js (hunk 1 above)
Comment (michitux) [reply xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/tests/unit/panels/LivedataAdvancedPanelFilter.spec.js]: Not with this structure, but with the split test method it's possible so I've changed it.

**Pull request merged**